### ISSUE TITLE
How to freeze parameters other than the backbone in MMDetection?

### ISSUE THREAD

**Issue by marcm09** [open-mmlab/mmdetection#10804]
I am trying to finetune models for instance segmentation such as mask-rcnn or cascade mask-rcnn.

I am building my configuration file according to the documentation for the mmdetection library. In this documentation, it is mentioned that the parameters should be frozen except for the last few layers.

However, it is not the case in my code since after specifying for the backbone the number of frozen stages (which was not mentioned as required in the documentation), the training time decreased drastically. Hence, only the backbone parameters are frozen currently and I also want to freeze all parameters except the ones for bbox_head and mask_head.

Similar issues were found for a previous version of mmdetection, and the fix was to change the list of parameters in the source code, however, in this version, this solution does not work since the source code changed.

**Comment by MEYE66** [comment]
Any update?

**Comment by ducnt1210** [comment]
Maybe you can specify which part of the model you want to freeze and set their `requires_grad = False` like this
```
for param in self.neck.parameters():
    param.requires_grad = False
for m in self.neck.modules():
    if isinstance(m, (nn.BatchNorm2d, nn.GroupNorm, _BatchNorm)):
        m.eval()
```
and remember to overwrite the train() function of the model because it is being called every epoch
```
def train(self, mode=True):
      super().train(mode)
      for param in self.neck.parameters():
          param.requires_grad = False
      for m in self.neck.modules():
          if isinstance(m, (nn.BatchNorm2d, nn.GroupNorm, _BatchNorm)):
		m.eval()
      return self
```
